include/zephyr/drivers/adc.h

```diff
@@ -218,7 +218,7 @@ struct adc_channel_cfg {
 	.channel_id       = DT_REG_ADDR(node_id), \
 IF_ENABLED(CONFIG_ADC_CONFIGURABLE_INPUTS, \
 	(.differential    = DT_NODE_HAS_PROP(node_id, zephyr_input_negative), \
-	 .input_positive  = DT_PROP_OR(node_id, zephyr_input_positive, 0), \
+	 .input_positive  = DT_PROP(node_id, zephyr_input_positive), \
```

Comment:
This will not work when an ADC driver that uses configurable inputs is used together with one that does not. See #52914.

Reply:
@anangl Maybe then it would be better to indicate configurable inputs as a devicetree property instead of a Kconfig?